registry.go

```diff
@@ -0,0 +1,239 @@
+package skynet
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"gitlab.com/NebulousLabs/encoding"
+	"gitlab.com/NebulousLabs/errors"
+	"golang.org/x/crypto/ed25519"
+	"net/http"
+	"net/url"
+)
+
+const (
+	// RegistryEndpoint is the registry endpoint
+	RegistryEndpoint = "/skynet/registry"
+	ed25519Algorithm = "ed25519"
+)
+
+type (
+	// RegistryEntryResponse contains the response of the registry entry call.
+	RegistryEntryResponse struct {
+		// Data contains the stored data in the entry.
+		Data string `json:"data"`
+		// Revision is the revision number for the entry.
+		Revision uint64 `json:"revision"`
+		// Signature is the signature of the registry entry.
+		Signature string `json:"signature"`
+	}
+
+	// RegistryEntry is the content of the registry entry.
+	RegistryEntry struct {
+		// DataKey is the key of the data for the given entry.
+		DataKey string
+		// Data contains the stored data in the entry.
+		Data string
+		// Revision is the revision number for the entry.
+		Revision uint64
+	}
+
+	// SignedRegistryEntry is the signed registry entry.
+	SignedRegistryEntry struct {
+		// Entry is the content of the registry entry.
+		Entry RegistryEntry
+		// Signature is the signature of the registry entry.
+		Signature Signature
+	}
+
+	// SetEntryPublicKey contains information about registry entry publicKey.
+	SetEntryPublicKey struct {
+		// Algorithm is the used algorithm
+		Algorithm string `json:"algorithm"`
+		// Key is the publicKey.
+		Key []byte `json:"key"`
+	}
+
+	// SetEntryRequestBody is the body content used to set registry entry.
+	SetEntryRequestBody struct {
+		// Publickey contains information about registry entry publicKey.
+		Publickey SetEntryPublicKey `json:"publickey"`
+		// DataKey is the key of the data for the given entry.
+		Datakey string `json:"datakey"`
+		// Revision is the revision number for the entry.
+		Revision int `json:"revision"`
+		// Data contains the stored data in the entry.
+		Data []byte `json:"data"`
+		// Signature is the signature of the registry entry.
+		Signature Signature `json:"signature"`
+	}
+
+	// Signature proves that data was signed by the owner of a particular
+	// public key's corresponding secret key.
+	Signature [ed25519.SignatureSize]byte
+)
+
+// GetEntry gets the registry entry corresponding to the publicKey and dataKey.
+func (sc *SkynetClient) GetEntry(
+	publicKey string,
+	dataKey string,
+) (r RegistryEntryResponse, err error) {
+	// TODO: use timeout
+	dataKeyHash := hashDataKey(dataKey)
+
+	values := url.Values{}
+	values.Set("publickey", fmt.Sprintf("%s:%s", ed25519Algorithm, publicKey))
+	values.Set("datakey", hex.EncodeToString(dataKeyHash))
+
+	resp, err := sc.executeRequest(
+		requestOptions{
+			Options:   sc.Options,
+			method:    "GET",
+			reqBody:   &bytes.Buffer{},
+			extraPath: RegistryEndpoint,
+			query:     values,
+		},
+	)
+	if err != nil {
+		return r, errors.AddContext(err, "could not execute request")
```

Comment:
Should return `null` instead of an error on 404 (entry not found) - see `skynet-js`.